scap kernel module fails to build on Linux >= 6.13

**Describe the bug**

scap fails to build. Here's a bug report showing the failure with falcosecurity-libs 0.18.1: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1093098

I just tried with 0.20.0, and I see the same failure

**How to reproduce it**

Build against a recent-enough kernel

**Environment**

Debian amd64.

Thanks.

Hi! 
I think the build issues you are facing come from:
```
[configure-kmod] Including /var/lib/dkms/scap/0.18.1/build//configure/ACCESS_OK_2/Makefile.inc /var/lib/dkms/scap/0.18.1/build//configure/CLASS_CREATE_1/Makefile.inc /var/lib/dkms/scap/0.18.1/build//configure/DEVNODE_ARG1_CONST/Makefile.inc
[configure-kmod] Build output for HAS_ACCESS_OK_2:
[configure-kmod] make: Entering directory '/var/lib/dkms/scap/0.18.1/build/configure/ACCESS_OK_2' make -C /var/lib/dkms/scap/0.18.1/build M=/var/lib/dkms/scap/0.18.1/build/configure/ACCESS_OK_2 modules make[1]: Entering directory '/var/lib/dkms/scap/0.18.1/b
uild' make[1]: *** No rule to make target 'modules'.  Stop. make[1]: Leaving directory '/var/lib/dkms/scap/0.18.1/build' make: *** [Makefile:15: all] Error 2 make: Leaving directory '/var/lib/dkms/scap/0.18.1/build/configure/ACCESS_OK_2'
[configure-kmod] Build output for HAS_CLASS_CREATE_1:
[configure-kmod] make: Entering directory '/var/lib/dkms/scap/0.18.1/build/configure/CLASS_CREATE_1' make -C /var/lib/dkms/scap/0.18.1/build M=/var/lib/dkms/scap/0.18.1/build/configure/CLASS_CREATE_1 modules make[1]: Entering directory '/var/lib/dkms/scap/0.
18.1/build' make[1]: *** No rule to make target 'modules'.  Stop. make[1]: Leaving directory '/var/lib/dkms/scap/0.18.1/build' make: *** [Makefile:15: all] Error 2 make: Leaving directory '/var/lib/dkms/scap/0.18.1/build/configure/CLASS_CREATE_1'
[configure-kmod] Build output for HAS_DEVNODE_ARG1_CONST:
[configure-kmod] make: Entering directory '/var/lib/dkms/scap/0.18.1/build/configure/DEVNODE_ARG1_CONST' make -C /var/lib/dkms/scap/0.18.1/build M=/var/lib/dkms/scap/0.18.1/build/configure/DEVNODE_ARG1_CONST modules make[1]: Entering directory '/var/lib/dkms
/scap/0.18.1/build' make[1]: *** No rule to make target 'modules'.  Stop. make[1]: Leaving directory '/var/lib/dkms/scap/0.18.1/build' make: *** [Makefile:15: all] Error 2 make: Leaving directory '/var/lib/dkms/scap/0.18.1/build/configure/DEVNODE_ARG1_CONST'
```

> No rule to make target 'modules'

Basically, the `build/configure/*` modules are small kernel modules that gets built as part of the kmod build process to check whether various changes are present or not (mostly in case of backports where we cannot rely upon kernel version macros); here they are: https://github.com/falcosecurity/libs/tree/master/driver/configure

In your case, all these modules fail to build thus the driver build proceeds as if all of these checks failed, not enabling the usage of correct APIs.

Am not sure about what's causing the `No rule to make target 'modules'` failure though.

I also encountered an error when compiling the ebpf probe: 

```
sysdig --bpf "(proc.name contains filter or proc.name in (bash))"
* Running scap-driver-loader for: driver version=0.18.1, arch=x86_64, kernel release=6.13.3-zen1-1-zen, kernel version=1
* Running scap-driver-loader with: driver=bpf, compile=yes, download=yes
* Filename 'scap_arch_6.13.3-zen1-1-zen_1.o' is composed of:
 - driver name: scap
 - target identifier: arch
 - kernel release: 6.13.3-zen1-1-zen
 - kernel version: 1
* Trying to download a prebuilt eBPF probe from https://download.sysdig.com/scap-drivers/0.18.1/x86_64/scap_arch_6.13.3-zen1-1-zen_1.o
curl: (22) The requested URL returned error: 404
Unable to find a prebuilt scap eBPF probe
* Trying to compile the eBPF probe (scap_arch_6.13.3-zen1-1-zen_1.o)
expr: syntax error: unexpected argument '1'
In file included from probe.c:17:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/include/linux/sched.h:12:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/current.h:10:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/include/linux/cache.h:6:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/cache.h:5:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/include/linux/linkage.h:8:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/linkage.h:6:
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/ibt.h:77:8: warning: 'nocf_check' attribute ignored; use -fcf-protection to enable the attribute [-Wignored-attributes]
   77 | extern __noendbr u64 ibt_save(bool disable);
      |        ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/ibt.h:32:34: note: expanded from macro '__noendbr'
   32 | #define __noendbr       __attribute__((nocf_check))
      |                                        ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/ibt.h:78:8: warning: 'nocf_check' attribute ignored; use -fcf-protection to enable the attribute [-Wignored-attributes]
   78 | extern __noendbr void ibt_restore(u64 save);
      |        ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/ibt.h:32:34: note: expanded from macro '__noendbr'
   32 | #define __noendbr       __attribute__((nocf_check))
      |                                        ^
In file included from probe.c:17:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/include/linux/sched.h:12:
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/current.h:47:10: warning: multiple identical address spaces specified for type [-Wduplicate-decl-specifier]
   47 |                 return this_cpu_read_const(const_pcpu_hot.current_task);
      |                        ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:578:36: note: expanded from macro 'this_cpu_read_const'
  578 | #define this_cpu_read_const(pcp)                        __raw_cpu_read_const(pcp)
      |                                                         ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:163:35: note: expanded from macro '__raw_cpu_read_const'
  163 | #define __raw_cpu_read_const(pcp)       __raw_cpu_read(, , pcp)
      |                                         ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:155:30: note: expanded from macro '__raw_cpu_read'
  155 |         *(qual __my_cpu_type(pcp) *)__my_cpu_ptr(&(pcp));               \
      |                                     ^
note: (skipping 1 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:94:40: note: expanded from macro '__my_cpu_type'
   94 | #define __my_cpu_type(var)      typeof(var) __percpu_seg_override
      |                                             ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:45:32: note: expanded from macro '__percpu_seg_override'
   45 | # define __percpu_seg_override  __seg_gs
      |                                 ^
<built-in>:366:33: note: expanded from macro '__seg_gs'
  366 | #define __seg_gs __attribute__((address_space(256)))
      |                                 ^
In file included from probe.c:17:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/include/linux/sched.h:12:
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/current.h:47:10: warning: multiple identical address spaces specified for type [-Wduplicate-decl-specifier]
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:578:36: note: expanded from macro 'this_cpu_read_const'
  578 | #define this_cpu_read_const(pcp)                        __raw_cpu_read_const(pcp)
      |                                                         ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:163:35: note: expanded from macro '__raw_cpu_read_const'
  163 | #define __raw_cpu_read_const(pcp)       __raw_cpu_read(, , pcp)
      |                                         ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:155:9: note: expanded from macro '__raw_cpu_read'
  155 |         *(qual __my_cpu_type(pcp) *)__my_cpu_ptr(&(pcp));               \
      |                ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:94:40: note: expanded from macro '__my_cpu_type'
   94 | #define __my_cpu_type(var)      typeof(var) __percpu_seg_override
      |                                             ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:45:32: note: expanded from macro '__percpu_seg_override'
   45 | # define __percpu_seg_override  __seg_gs
      |                                 ^
<built-in>:366:33: note: expanded from macro '__seg_gs'
  366 | #define __seg_gs __attribute__((address_space(256)))
      |                                 ^
In file included from probe.c:17:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/include/linux/sched.h:13:
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/processor.h:563:10: warning: multiple identical address spaces specified for type [-Wduplicate-decl-specifier]
  563 |                 return this_cpu_read_const(const_pcpu_hot.top_of_stack);
      |                        ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:578:36: note: expanded from macro 'this_cpu_read_const'
  578 | #define this_cpu_read_const(pcp)                        __raw_cpu_read_const(pcp)
      |                                                         ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:163:35: note: expanded from macro '__raw_cpu_read_const'
  163 | #define __raw_cpu_read_const(pcp)       __raw_cpu_read(, , pcp)
      |                                         ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:155:30: note: expanded from macro '__raw_cpu_read'
  155 |         *(qual __my_cpu_type(pcp) *)__my_cpu_ptr(&(pcp));               \
      |                                     ^
note: (skipping 1 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:94:40: note: expanded from macro '__my_cpu_type'
   94 | #define __my_cpu_type(var)      typeof(var) __percpu_seg_override
      |                                             ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:45:32: note: expanded from macro '__percpu_seg_override'
   45 | # define __percpu_seg_override  __seg_gs
      |                                 ^
<built-in>:366:33: note: expanded from macro '__seg_gs'
  366 | #define __seg_gs __attribute__((address_space(256)))
      |                                 ^
In file included from probe.c:17:
In file included from /usr/lib/modules/6.13.3-zen1-1-zen/build/include/linux/sched.h:13:
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/processor.h:563:10: warning: multiple identical address spaces specified for type [-Wduplicate-decl-specifier]
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:578:36: note: expanded from macro 'this_cpu_read_const'
  578 | #define this_cpu_read_const(pcp)                        __raw_cpu_read_const(pcp)
      |                                                         ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:163:35: note: expanded from macro '__raw_cpu_read_const'
  163 | #define __raw_cpu_read_const(pcp)       __raw_cpu_read(, , pcp)
      |                                         ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:155:9: note: expanded from macro '__raw_cpu_read'
  155 |         *(qual __my_cpu_type(pcp) *)__my_cpu_ptr(&(pcp));               \
      |                ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:94:40: note: expanded from macro '__my_cpu_type'
   94 | #define __my_cpu_type(var)      typeof(var) __percpu_seg_override
      |                                             ^
/usr/lib/modules/6.13.3-zen1-1-zen/build/arch/x86/include/asm/percpu.h:45:32: note: expanded from macro '__percpu_seg_override'
   45 | # define __percpu_seg_override  __seg_gs
      |                                 ^
<built-in>:366:33: note: expanded from macro '__seg_gs'
  366 | #define __seg_gs __attribute__((address_space(256)))
      |                                 ^
In file included from probe.c:27:
./fillers.h:873:56: error: member reference base type 'struct percpu_counter[4]' is not a structure or union
  873 |         bpf_probe_read_kernel(&val, sizeof(val), &mm->rss_stat.count[member]);
      |                                                   ~~~~~~~~~~~~^~~~~~
./fillers.h:2285:48: warning: passing 'volatile long *' to parameter of type 'long *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
 2285 |                 res = bpf_accumulate_argv_or_env(data, argv, &args_len);
      |                                                              ^~~~~~~~~
./fillers.h:1895:61: note: passing argument to parameter 'args_len' here
 1895 |                                                       long *args_len) {
      |                                                             ^
7 warnings and 1 error generated.
make[4]: *** [Makefile:74: probe.o] Error 1
make[3]: *** [/usr/lib/modules/6.13.3-zen1-1-zen/build/Makefile:1987: .] Error 2
make[2]: *** [/usr/lib/modules/6.13.3-zen1-1-zen/build/Makefile:251: __sub-make] Error 2
make[1]: *** [Makefile:251: __sub-make] Error 2
make: *** [Makefile:23: all] Error 2
mv: cannot stat '/usr/src/scap-0.18.1/bpf/probe.o': No such file or directory
Unable to load the scap eBPF probe
Unable to load the BPF probe
BPF probe is compiled for 6.11.9-zen1-1-zen, but running version is 6.13.3-zen1-1-zen

```

@unknowndevQwQ 
> /fillers.h:873:56: error: member reference base type 'struct percpu_counter[4]' is not a structure or union
  873 |         bpf_probe_read_kernel(&val, sizeof(val), &mm->rss_stat.count[member]);

It seems like the correct configure module (https://github.com/falcosecurity/libs/tree/master/driver/bpf/configure/RSS_STAT_ARRAY in this case) was not run prior to the build of the eBPF probe.
cc @therealbobo

--- a/driver/Makefile.in
+++ b/driver/Makefile.in
@@ -29,7 +29,7 @@ install: all
 
 else
 
-KERNELDIR 	?= $(CURDIR)
+KERNELDIR 	?= $(realpath $(objtree))
 #
 # Get the path of the module sources
 #

--- a/driver/bpf/Makefile
+++ b/driver/bpf/Makefile
@@ -28,7 +28,7 @@ clean:
 
 else
 
-KERNELDIR 	?= $(CURDIR)
+KERNELDIR 	?= $(realpath $(objtree))
 #
 # Get the path of the module sources
 #

